### ISSUE TITLE
Gframe

### ISSUE THREAD

**Issue by ghost** [hamhub7/nx-TAS#15]
This bug causes a frame during a tas to be "held down" leaving the rest of the script unable to be executed. I have had both a joystick input and a Z button gframe so far, but it is probable there is more.

To stop this bug, you must restart console. Until console is restarted, no new commands can be executed using nx-tas; no scripts, and no adding/removing controllers.

You can also remove the nx-tas controller through the controller menu but the above issues persist (nothing else can be done in nx tas until console is restarted). I have also had the issue stop after just leaving my switch for half an hour, but again, no nx tas until console restart.

I have had this issue both with a procon and joycons as the nx-tas controllers, and both with the SMO practice mod installed and active, and installed and inactive.

This issue is not an issue with SMO, as I have this issue wherever I execute scripts, be it in SMO, Zelda, or just the home menu.

This may be a thermal issue, as Soren suggested, but I have had it happen when it is 5 degrees celsius outside. CMFP also reports experiencing this issue.

I have below some files where Gframe does and doesn't occur. (Pastebin cause I already had them handy)

The glitch occurs in;
https://pastebin.com/ien8t9xW (This is my seaside script. This can cause either the joystick to be held to the right after a spinjump or mario to permanently crouch.)

It also occurs here;
https://pastebin.com/MHhA624s (**Removing the last 30 frames of this script will cause Gframe to not occur** - see https://pastebin.com/ihBjH7aj. This makes me think Gframe could be to do with specific angles on a joystick or similar.)

Video examples;
https://twitter.com/bork_mc/status/1259923405832245249 - script messed up due to loads but mario is not supposed to be crouching
https://twitter.com/bork_mc/status/1259128396287586305 - script was only 100 frames long yet kept holding forward, repeatedly killing mario

**Comment by redanaheim** [comment]
Interestingly, this issue is also consistent with the same script, even if the script has no errors whatsoever.

**Comment by s5bug** [comment]
I have a hunch as to why this might happen.

What happens when you add or remove an empty line at the end of your script? I.e. is there a difference between
```
N KEY_FOO X;Y Z;W
```
and
```
N KEY_FOO X;Y Z;W

```
?

**Comment by redanaheim** [comment]
Nope, this doesn't change anything. The whole time I was just experimenting, the script I was using to compile doesn't add an empty line at the bottom, and it still happened sometimes. Plus the issue can happen in the middle of scripts.

**Comment by s5bug** [comment]
I'd like to see if the same thing happens with sys-script's implementation of TASing. I'll write up a small guide later today of how to test it.

**Comment by ghost** [comment]
https://twitter.com/bork_mc/status/1260182711601258496

here's another example of the glitch

**Comment by s5bug** [comment]
Does the ghost frame happen in scripts without joystick inputs? I.e. does it happen when only using buttons and leaving joysticks as 0;0?

**Comment by s5bug** [comment]
1. Disable nx-TAS by removing `sdmc:/atmosphere/contents/0100000000000cab/flags/boot2.flag`
2. Install sys-script by placing the attached NSP as `sdmc:/atmosphere/contents/4200736372697074/exefs.nsp`
3. Enable sys-script by creating an empty file as `sdmc:/atmosphere/contents/4200736372697074/flags/boot2.flag`
4. Save [this TAS library](https://mystb.in/defebixape.lisp) as `sdmc:/scripts/tas.janet` (As of this posting, said script **does not have most buttons** and **does not have any joystick support**. It will be updated shortly. Additionally, **it presses L+R, and then A before actually starting the script**.)
5. Create a new file at `sdmc:/scripts/script1.janet` that looks like [this](https://mystb.in/egupilecot.lisp) (replacing `"sdmc:/scripts/script8.txt"` with your nx-TAS script's path)
6. Press F1 whenever you need to execute `script1.janet`

You can modify the library you download in step 4 pretty easily if you know Lisp or can understand it at a glance.

[sys-script.zip](https://github.com/hamhub7/nx-TAS/files/4630117/sys-script.zip)

**Comment by s5bug** [comment]
The scripts linked in [the first post of this issue](https://github.com/hamhub7/nx-TAS/issues/15#issue-616115563) are missing right joystick values, and as such, will rightfully crash.

Whether well-formed scripts still exhibit this behavior is under investigation.

**Comment by redanaheim** [comment]
Steps for migrating from nx-TAS to sys-script:
1. Remove `boot2.flag` in `sdmc:/atmosphere/contents/0100000000000cab`.
2. `git clone https://github.com/sorenbug/sys-script.git`, `cd sys-script`, `make`.
3. Run `make sd` in the sys-script main directory. In the newly created `sd` folder, enter `atmosphere/contents` and drag that into your `atmosphere/contents` on your SD. Drag the `script` folder in the `sd` folder to the root of your SD.
4. Download this `main.janet` (https://pastebin.com/4T9NyggF) and replace the one in `sdmc:/script/`.
5. You may have to turn on sys-script from Kosmos Toolbox -> Background Services on boot.
6. To add the controller (only 1 at a time), press =. 
7. Run scripts like normal by pressing F1 - F12 keys.

To reset if you are having gframe or the script got stuck because of an error, turn off sys-script in Kosmos Toolbox and turn it back on again. To check logs if you had an error, look at the contents of `sdmc:/script/log`.

**Comment by sorenbug** [comment]
You have to create `sdmc:/boot.janet.log` before sys-script will even _run_ at the current point in time.

**Comment by redanaheim** [comment]
Thanks, I've edited the comment

**Comment by ghost** [comment]
I'm going to close it as it seems like I was just being dumb.

**Comment by Expertsonicgamer32x** [comment]
have fun being jel about me getting what i want if thats why you kicked but besides you ham everbody else was awesome i might going to far but who cares you can delete this anyways ik you have to see it before you delete obviously so bu bye. and thanks for clone-nx

**Comment by Expertsonicgamer32x** [comment]
ha ik you were jel cause you deleted my other comment ok just wanted to say this you can delete it now alright now im totally done
